### ISSUE TITLE
Don't allow assigning return values to objects

### ISSUE THREAD

**Issue by jeremy-ebler-vineti** [cypress-io/eslint-plugin-cypress#20]
I have a team that's started writing Cypress like:

```javascript
context('Foo', () => {
  const scope = {}
  it('allows this', () => {
    scope.baddie = cy.get('it')
  })
  ...
})
```

I was expecting `no-assigning-return-values` to error out on the assignment of a cypress command to `scope.baddie`.

`const scope = {}` is also discouraged by [Cypress's Docs](https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html#Variables), but that's another issue.

**Comment by brettz9** [comment]
For whoever might work on implementing this change, there are plenty of other contexts where a return value gets used, e.g., within a template, as part of a binary expression, etc. etc.

In our work on `eslint-plugin-unicorn`, we found that rather than trying to blacklist the various other contexts where the return value might be used besides variable declarations (discussed more fully at https://github.com/sindresorhus/eslint-plugin-unicorn/pull/503 ), that a simple [utility](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/utils/is-value-not-usable.js) could be used to whitelist only those cases where the return values are known *not* to be used, i.e.:

```js
module.exports = ({parent}) => !parent || parent.type === 'ExpressionStatement';
```

Then just call this method with the `node` in question, e.g., `isValueNotUsable(node)` to see whether this is safe (and an error can be reported otherwise).

**Comment by brettz9** [comment]
By the way, this proposed expanding to other contexts could help catch another problem, namely that Cypress doesn't seem to like `await` to be used (though I wonder whether this can be liberalized to be permitted where a `Promise` is explicitly allowed, i.e., with `cy.task`).

**Comment by brettz9** [comment]
Or if #6575 is implemented, to liberalize returning a (Promise) value from a known custom command.

**Comment by MikeMcC399** [comment]
- https://github.com/cypress-io/cypress/issues/6575 was closed as not planned
